### ISSUE TITLE
Application not responding after registration

### ISSUE THREAD

**Issue by jakubdolejs** [AppliedRecognition/Ver-ID-Android-Sample#1]
### Steps to re-create

- After installing the application and starting registration part 3 photos are taken and a analyzing screen is shown and stays there, it does not proceed further exiting the application by pressing the home button and starting it again and trying to restart the registration part freezes application and ultimately a ANR is shown
- Tested the app on custom-made devices
- Using 2.0.5 version of lib
- No logs available when ANR occurs

**Comment by yatin009** [comment]
Following testing is done on Android 7.0, API level 24 using Ver-Id android sample app
from  https://github.com/AppliedRecognition/Ver-ID-Android-Sample

Two parts to this issue :

1. Application does not proceed from analyzing screen, which is right after taking photos for registration

screenshot of logcat

![locat_sanp](https://user-images.githubusercontent.com/13184840/35280612-b3256988-001e-11e8-89f5-5cdfe11a9245.PNG)

stacktrace
````
01-01 17:39:38.210 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:38.227 2429-2449/com.appliedrec.ver_idsample D/GraphicBuffer: register, handle(0x90cabcc0) (w:720 h:1280 s:720 f:0x1 u:0x000b00)
01-01 17:39:38.249 2429-2449/com.appliedrec.ver_idsample D/Surface: Surface::disconnect(this=0x91169b00,api=1)
01-01 17:39:38.251 2429-2449/com.appliedrec.ver_idsample D/GraphicBuffer: unregister, handle(0x90cabc00) (w:720 h:1280 s:720 f:0x1 u:0x000b00)
01-01 17:39:38.251 2429-2449/com.appliedrec.ver_idsample D/GraphicBuffer: unregister, handle(0x90cabe40) (w:720 h:1280 s:720 f:0x1 u:0x000b00)
01-01 17:39:38.251 2429-2449/com.appliedrec.ver_idsample D/GraphicBuffer: unregister, handle(0x90cabf00) (w:720 h:1280 s:720 f:0x1 u:0x000b00)
01-01 17:39:38.252 2429-2449/com.appliedrec.ver_idsample D/Surface: Surface::disconnect(this=0x91169b00,api=1)
01-01 17:39:38.275 2429-2429/com.appliedrec.ver_idsample I/SurfaceView: updateWindow -- setFrame, this = com.appliedrec.shared.ui.CameraSurfaceView{afc7ac8 V.E...... ......ID 0,-24-720,1256 #2}
01-01 17:39:38.279 2429-2429/com.appliedrec.ver_idsample I/SurfaceView: updateWindow -- onWindowVisibilityChanged, visibility = 8, this = com.appliedrec.shared.ui.CameraSurfaceView{afc7ac8 V.E...... ......ID 0,-24-720,1256 #2}
01-01 17:39:38.279 2429-2429/com.appliedrec.ver_idsample I/SurfaceView: 184318664 Changes: creating=false format=false size=false visible=true left=false top=false mUpdateWindowNeeded=false mReportDrawNeeded=false redrawNeeded=false forceSizeChanged=false mVisible=true mRequestedVisible=false, this = com.appliedrec.shared.ui.CameraSurfaceView{afc7ac8 V.E...... ......ID 0,-24-720,1256 #2}
01-01 17:39:38.279 2429-2429/com.appliedrec.ver_idsample I/SurfaceView: 184318664 Cur surface: Surface(name=null)/@0x68125b1, this = com.appliedrec.shared.ui.CameraSurfaceView{afc7ac8 V.E...... ......ID 0,-24-720,1256 #2}
01-01 17:39:38.283 2429-2589/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:38.291 2429-2429/com.appliedrec.ver_idsample I/SurfaceView: 184318664 New surface: Surface(name=null)/@0x65f2a96, vis=false, frame=Rect(0, 0 - 720, 1280), this = com.appliedrec.shared.ui.CameraSurfaceView{afc7ac8 V.E...... ......ID 0,-24-720,1256 #2}
01-01 17:39:38.291 2429-2429/com.appliedrec.ver_idsample I/SurfaceView: 184318664 visibleChanged -- surfaceDestroyed, this = com.appliedrec.shared.ui.CameraSurfaceView{afc7ac8 V.E...... ......ID 0,-24-720,1256 #2}
01-01 17:39:38.292 2429-2429/com.appliedrec.ver_idsample I/SurfaceView: surfaceDestroyed callback +, this = com.appliedrec.shared.ui.CameraSurfaceView{afc7ac8 V.E...... ......ID 0,-24-720,1256 #2}
01-01 17:39:38.292 2429-2429/com.appliedrec.ver_idsample I/SurfaceView: surfaceDestroyed callback -, this = com.appliedrec.shared.ui.CameraSurfaceView{afc7ac8 V.E...... ......ID 0,-24-720,1256 #2}
01-01 17:39:38.297 2429-2449/com.appliedrec.ver_idsample D/SurfaceView: 184318664 windowPositionLostRT RT, frameNr = 0
01-01 17:39:38.361 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:38.525 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:38.595 2429-2588/com.appliedrec.ver_idsample D/ImageUtil: +--------------------+
                                                                      |        0354        |
                                                                      |      +------+      |
                                                                      |      | 0520 |      |
                                                                      | 0071 |  x   | 0129 |
                                                                      |      | 0650 |      |
                                                                      |      +------+      |
                                                                      |        0276        |
                                                                      +--------------------+
01-01 17:39:38.650 2429-2429/com.appliedrec.ver_idsample D/VerIDFrgmt: Generated result with outcome SUCCESS
01-01 17:39:40.180 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:40.183 2429-2591/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:40.272 2429-2591/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:40.303 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:42.159 2429-2565/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:42.256 2429-2565/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:42.264 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:42.308 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:44.147 2429-2589/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:44.250 2429-2589/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:44.259 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:44.297 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:46.166 2429-2591/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:46.262 2429-2591/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:46.275 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:46.316 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:48.169 2429-2565/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:48.266 2429-2565/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:48.279 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:48.316 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:50.175 2429-2589/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:50.276 2429-2589/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:50.291 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:50.327 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:52.175 2429-2591/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:52.272 2429-2591/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:52.288 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:52.312 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:54.163 2429-2565/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:54.253 2429-2565/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:54.262 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:54.311 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:56.134 2429-2589/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:56.230 2429-2589/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:56.242 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:56.283 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:58.137 2429-2591/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:39:58.235 2429-2591/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:39:58.244 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:39:58.300 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:40:00.149 2429-2565/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:40:00.254 2429-2565/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:40:00.265 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:40:00.308 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:40:00.325 2429-2440/com.appliedrec.ver_idsample I/art: Background partial concurrent mark sweep GC freed 52(1736B) AllocSpace objects, 3(9MB) LOS objects, 14% free, 23MB/27MB, paused 10.484ms total 54.076ms
01-01 17:40:02.125 2429-2589/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels +
01-01 17:40:02.216 2429-2589/com.appliedrec.ver_idsample D/skia: SkJpegCodec::onGetPixels -
01-01 17:40:02.225 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
01-01 17:40:02.268 2429-2440/com.appliedrec.ver_idsample I/art: Enter while loop.
````
2. After restarting the application

Application keeps on loading Ver-Id

![image uploaded from ios](https://user-images.githubusercontent.com/13184840/35281038-e9b4eb1c-001f-11e8-8570-816ef1bf2835.jpg)

logcat screenshot

![logcat_snap_2](https://user-images.githubusercontent.com/13184840/35281060-f8d0643c-001f-11e8-8962-fdb088278242.PNG)

stacktrace

```
01-01 17:43:17.427 2607-2625/com.appliedrec.ver_idsample D/OpenGLRenderer: ProgramCache.generateProgram: 5242945
01-01 17:43:17.428 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.427 CORE35   INFO: thread enter
01-01 17:43:17.428 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.427 0x99fdf5c0 INFO: pca::initialize_context: settings {
01-01 17:43:17.429 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.428 CORE37   INFO: thread enter
01-01 17:43:17.429 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.427 CORE36   INFO: thread enter
01-01 17:43:17.430 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.429 CORE40   INFO: thread enter
01-01 17:43:17.430 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.429 CORE38   INFO: thread enter
01-01 17:43:17.430 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.430 CORE41   INFO: thread enter
01-01 17:43:17.431 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.429 0x99fdf5c0 INFO: pca::initialize_context: settings {
01-01 17:43:17.431 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.429 CORE39   INFO: thread enter
01-01 17:43:17.431 2607-2625/com.appliedrec.ver_idsample D/OpenGLRenderer: ProgramCache.generateProgram: 240518168576
01-01 17:43:17.432 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.431 0x99fdf5c0 INFO: pca::initialize_context: settings {
01-01 17:43:17.432 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.432 CORE42   INFO: thread enter
01-01 17:43:17.433 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.432 0x99fdf5c0 INFO: pca::initialize_context: settings {
01-01 17:43:17.434 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.433 CORE44   INFO: thread enter
01-01 17:43:17.434 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.433 0x99fdf5c0 INFO: det::prepare_detection: start
01-01 17:43:17.434 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.433 CORE43   INFO: thread enter
01-01 17:43:17.436 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.434 CORE45   INFO: thread enter
01-01 17:43:17.436 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.434 CORE46   INFO: thread enter
01-01 17:43:17.436 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.435 0x99fdf5c0 INFO: dlib face detector: "/data/user/0/com.appliedrec.ver_idsample/files/models/fhogcascade_face_frontal.dat"
01-01 17:43:17.437 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.435 CORE47   INFO: thread enter
01-01 17:43:17.441 2607-2625/com.appliedrec.ver_idsample D/GraphicBuffer: register, handle(0xad31f240) (w:720 h:1280 s:720 f:0x1 u:0x000b00)
01-01 17:43:17.472 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:17.471 0x99fdf5c0 INFO: dlib object: "/data/user/0/com.appliedrec.ver_idsample/files/models/shape_predictor_68_face_landmarks.dat"
01-01 17:43:17.477 2607-2625/com.appliedrec.ver_idsample D/GraphicBuffer: register, handle(0xad31f300) (w:720 h:1280 s:720 f:0x1 u:0x000b00)
01-01 17:43:20.355 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:20.353 CORE2    INFO: dlib face detector: frontal
01-01 17:43:20.355 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:20.353 0x99fdf5c0 INFO: dlib face detector: frontal
01-01 17:43:20.355 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:20.353 CORE1    INFO: dlib face detector: frontal
01-01 17:43:20.356 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:20.353 CORE0    INFO: dlib face detector: frontal
01-01 17:43:20.365 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:20.364 0x99fdf5c0 INFO: det::prepare_detection: done
01-01 17:43:20.365 2607-2636/com.appliedrec.ver_idsample I/DetRecLib: - 17:43:20.365 0x99fdf5c0 INFO: loading: "/data/user/0/com.appliedrec.ver_idsample/files/models/dlib_face_recognition_resnet_model_v1.dat"
01-01 17:43:21.401 2607-2636/com.appliedrec.ver_idsample D/DetRecLib: - 17:43:21.400 0x99fdf5c0 DEBUG: loading all encryption keys into map
01-01 17:43:21.402 2607-2636/com.appliedrec.ver_idsample D/DetRecLib: - 17:43:21.401 0x99fdf5c0 DEBUG: loading encryption key alias: DetRecLib Android AES/GCM/NoPadding
```

**Comment by jakubdolejs** [comment]
The bug was related to template storage encryption on devices where encryption is not available. Fixed in version 2.0.6.
